CONTRIBUTING.md

```diff
@@ -1,217 +1,155 @@
+```
 
-### Intel MAC
+### Updating the config files
+Service config files are also generated from code similarly to OpenAPI specs. When changing any of the service configuration data types, they have to be regeneraetd otherwise the CI process fails and asks for doing it.
 
-Typically, the following should allow you to run it successfully.
+To regenerate these files, use
 
-```bash
-brew tap messense/macos-cross-toolchains
-brew install messense/macos-cross-toolchains/x86_64-unknown-linux-gnu
-# If openssl is not in system
-# brew install openssl
-export OPENSSL_DIR=$(brew --prefix openssl)
-export CC_X86_64_UNKNOWN_LINUX_GNU=x86_64-unknown-linux-gnu-gcc
-export CXX_X86_64_UNKNOWN_LINUX_GNU=x86_64-unknown-linux-gnu-g++
-export AR_X86_64_UNKNOWN_LINUX_GNU=x86_64-unknown-linux-gnu-ar
-export CARGO_TARGET_X86_64_UNKNOWN_LINUX_GNU_LINKER=x86_64-unknown-linux-gnu-gcc
+```shell
+cargo make generate-configs
 ```
 
-From the root of the project
+### Preparing the pull request
+Golem CI checks the pull requests with `rustfmt` and `cargo clippy`. To make sure these checks pass, before opening a pull request run
 
-```bash
-rustup target add x86_64-unknown-linux-gnu
-cargo build --release --target x86_64-unknown-linux-gnu --package golem-shard-manager
-cargo build --release --target x86_64-unknown-linux-gnu --package golem-component-service
-cargo build --release --target x86_64-unknown-linux-gnu --package golem-worker-service
-cargo build --release --target x86_64-unknown-linux-gnu --package golem-component-compilation-service
-cargo build --release --target x86_64-unknown-linux-gnu --package golem-worker-executor
+```shell
+cargo make fix
 ```
 
-### ARM MAC
+and fix any possible errors and warnings reported by it.
 
-Typically, the following should allow you to run it successfully.
+## Running Golem locally
 
-```bash
-brew tap messense/macos-cross-toolchains
-brew install aarch64-unknown-linux-gnu
-# If openssl is not in system
-# brew install openssl
-export OPENSSL_DIR=$(brew --prefix openssl)
-export CC_AARCH64_UNKNOWN_LINUX_GNU=aarch64-unknown-linux-gnu-gcc
-export CXX_AARCH64_UNKNOWN_LINUX_GNU=aarch64-unknown-linux-gnu-g++
-export AR_AARCH64_UNKNOWN_LINUX_GNU=aarch64-unknown-linux-gnu-ar
-export CARGO_TARGET_AARCH64_UNKNOWN_LINUX_GNU_LINKER=aarch64-unknown-linux-gnu-gcc
-```
+There are two ways now to run Golem locally:
+
+### Using cargo make run
 
-From the root of the project
+By running
 
-```bash
-rustup target add aarch64-unknown-linux-gnu
-cargo build --release --target aarch64-unknown-linux-gnu --package golem-shard-manager
-cargo build --release --target aarch64-unknown-linux-gnu --package golem-component-service
-cargo build --release --target aarch64-unknown-linux-gnu --package golem-worker-service
-cargo build --release --target aarch64-unknown-linux-gnu --package golem-component-compilation-service
-cargo build --release --target aarch64-unknown-linux-gnu --package golem-worker-executor
+```shell
+cargo make run
```

Comment:
I was expecting this command to be on the top. But may be this is the right place

Reply:
I don't know I almost never run it while developing :)